src/main/kotlin/com/chattriggers/ctjs/minecraft/wrappers/world/Chunk.kt

```diff
@@ -58,51 +62,52 @@ class Chunk(override val mcValue: MCChunk) : CTWrapper<MCChunk> {
      */
     // TODO: Move this method somewhere else
     fun getBlockLightLevel(x: Int, y: Int, z: Int): Int {
-        return UMinecraft.getMinecraft().world?.getLightLevel(LightType.BLOCK, BlockPos(x, y, z)) ?: 0
+        return UMinecraft.getMinecraft().world?.getLightLevel(LightType.BLOCK, MCBlockPos(x, y, z)) ?: 0
     }
 
-    // TODO:
-    // /**
-    //  * Gets every entity in this chunk
-    //  *
-    //  * @return the entity list
-    //  */
-    // fun getAllEntities(): List<Entity> {
-    //     return chunk.entityLists.toList().flatten().map(::Entity)
-    // }
-    //
-    // /**
-    //  * Gets every entity in this chunk of a certain class
-    //  *
-    //  * @param clazz the class to filter for (Use `Java.type().class` to get this)
-    //  * @return the entity list
-    //  */
-    // fun getAllEntitiesOfType(clazz: Class<*>): List<Entity> {
-    //     return getAllEntities().filter {
-    //         clazz.isInstance(it.entity)
-    //     }
-    // }
+    /**
+     * Gets every entity in this chunk
+     *
+     * @return the entity list
+     */
+    fun getAllEntities(): List<Entity> {
+        val box = Box(MCBlockPos(getMinBlockX(), mcValue.bottomY, getMinBlockZ())).stretch(16.0, mcValue.topY.toDouble(), 16.0)
+
+        return World.toMC()?.getEntitiesByClass(MCEntity::class.java, box) { true }?.map(Entity::fromMC) ?: listOf()
+    }
+
+    /**
+     * Gets every entity in this chunk of a certain class
+     *
+     * @param clazz the class to filter for (Use `Java.type().class` to get this)
+     * @return the entity list
+     */
+    fun getAllEntitiesOfType(clazz: Class<*>): List<Entity> {
+        return getAllEntities().filter {
+            clazz.isInstance(it.toMC())
+        }
+    }
```

Comment:
Since `getEntitiesByClass` takes the class already, it might be nice to move the functionality in `getAllEntities` to a helper function that takes a `Class` and then call that from both